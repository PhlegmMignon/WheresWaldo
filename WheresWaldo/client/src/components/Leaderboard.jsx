export default function Leaderboard() {
  return <div id="leaderboard"></div>;
}
