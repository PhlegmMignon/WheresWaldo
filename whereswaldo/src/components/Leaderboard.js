import React from "react";

export default function Leaderboard(props) {
  const CARD_STYLE = {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
  };

  return (
    <div id="leaderboard">
      Leaderboard
      {props.scoreList.map((item) => {
        return (
          <div className="card" style={CARD_STYLE}>
            <div className="names">{item.name}</div>
            <div className="scores">{item.score / 100}s</div>
          </div>
        );
      })}
    </div>
  );
}
