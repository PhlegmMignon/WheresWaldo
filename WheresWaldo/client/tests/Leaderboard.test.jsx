import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";
import Leaderboard from "../src/components/Leaderboard.jsx";

describe("Leaderboard unit tests", () => {
  it("Renders stage select", () => {
    const { container } = render(
      <Leaderboard
        scores={[
          { name: "asd", score: 212, _id: "0" },
          { name: "qwe", score: 80, _id: "1" },
          { name: "zxc", score: 123, _id: "2" },
        ]}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

describe("Leaderboard integration tests", () => {
  it("Clicking new game calls updateState", async () => {
    const user = userEvent.setup();
    const fakeSetGameState = vi.fn();

    render(
      <Leaderboard
        scores={[
          { name: "asd", score: 212, _id: "0" },
          { name: "qwe", score: 80, _id: "1" },
          { name: "zxc", score: 123, _id: "2" },
        ]}
        setGameState={fakeSetGameState}
      />
    );

    const btn = screen.getByRole("button", { name: "New game" });
    await user.click(btn);

    expect(fakeSetGameState).toHaveBeenCalled();
  });
});
