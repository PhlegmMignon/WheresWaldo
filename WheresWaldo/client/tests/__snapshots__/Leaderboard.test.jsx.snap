// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Leaderboard unit tests > Renders stage select 1`] = `
<div>
  <div
    class="absolute inset-1/2 flex flex-col gap-3 h-72 w-60 rounded-xl p-6"
    id="leaderboard"
  >
    <div
      class="text-center text-lg "
    >
      Leaderboard
    </div>
    <div
      class="flex justify-between"
    >
      <p>
        Name
      </p>
       
      <p
        class="-ml-8"
      >
        Time
      </p>
    </div>
    <ol>
      <li
        class="flex justify-between"
      >
        <div>
          asd
        </div>
        <div>
          3
          mins 
          32
          secs
        </div>
      </li>
      <li
        class="flex justify-between"
      >
        <div>
          qwe
        </div>
        <div>
          1
          mins 
          20
          secs
        </div>
      </li>
      <li
        class="flex justify-between"
      >
        <div>
          zxc
        </div>
        <div>
          2
          mins 
          3
          secs
        </div>
      </li>
    </ol>
    <button
      class="text-lg"
    >
      New game
    </button>
  </div>
</div>
`;
